src/perform_step/gpu_vern7_perform_step.jl

```diff
@@ -0,0 +1,394 @@
+@inline function step!(integ::GPUVern7I{false, S, T}, ts, us) where {T, S}
+    @unpack dt = integ
+    t = integ.t
+    p = integ.p
+    @unpack c2, c3, c4, c5, c6, c7, c8, a021, a031, a032, a041, a043, a051, a053, a054,
+    a061, a063, a064, a065, a071, a073, a074, a075, a076, a081, a083, a084,
+    a085, a086, a087, a091, a093, a094, a095, a096, a097, a098, a101, a103,
+    a104, a105, a106, a107, b1, b4, b5, b6, b7, b8, b9 = integ.tab
+
+    tmp = integ.tmp
+    f = integ.f
+    integ.uprev = integ.u
+    uprev = integ.u
+
+    integ.tprev = t
+    saved_in_cb = false
+    adv_integ = true
+    ## Check if tstops are within the range of time-series
+    if integ.tstops !== nothing && integ.tstops_idx <= length(integ.tstops) &&
+       (integ.tstops[integ.tstops_idx] - integ.t - integ.dt - 100 * eps(integ.t) < 0)
+        integ.t = integ.tstops[integ.tstops_idx]
+        integ.tstops_idx += 1
+    else
+        ##Advance the integrator
+        integ.t += dt
+    end
+
+    if integ.u_modified
+        k1 = f(uprev, p, t)
+        integ.u_modified = false
+    else
+        @inbounds k1 = integ.k10
+    end
+
+    k1 = f(uprev, p, t)
+    a = dt * a021
+    k2 = f(uprev + a * k1, p, t + c2 * dt)
+    k3 = f(uprev + dt * (a031 * k1 + a032 * k2), p, t + c3 * dt)
+    k4 = f(uprev + dt * (a041 * k1 + a043 * k3), p, t + c4 * dt)
+    k5 = f(uprev + dt * (a051 * k1 + a053 * k3 + a054 * k4), p, t + c5 * dt)
+    k6 = f(uprev + dt * (a061 * k1 + a063 * k3 + a064 * k4 + a065 * k5), p, t + c6 * dt)
+    k7 = f(uprev + dt * (a071 * k1 + a073 * k3 + a074 * k4 + a075 * k5 + a076 * k6), p,
+           t + c7 * dt)
+    k8 = f(uprev +
+           dt * (a081 * k1 + a083 * k3 + a084 * k4 + a085 * k5 + a086 * k6 + a087 * k7), p,
+           t + c8 * dt)
+    g9 = uprev +
+         dt *
+         (a091 * k1 + a093 * k3 + a094 * k4 + a095 * k5 + a096 * k6 + a097 * k7 + a098 * k8)
+    g10 = uprev +
+          dt * (a101 * k1 + a103 * k3 + a104 * k4 + a105 * k5 + a106 * k6 + a107 * k7)
+    k9 = f(g9, p, t + dt)
+    k10 = f(g10, p, t + dt)
+
+    integ.u = uprev +
+              dt * (b1 * k1 + b4 * k4 + b5 * k5 + b6 * k6 + b7 * k7 + b8 * k8 + b9 * k9)
+
+    @unpack c11, a1101, a1104, a1105, a1106, a1107, a1108, a1109, c12, a1201, a1204,
+    a1205, a1206, a1207, a1208, a1209, a1211, c13, a1301, a1304, a1305, a1306, a1307,
+    a1308, a1309, a1311, a1312, c14, a1401, a1404, a1405, a1406, a1407, a1408, a1409,
+    a1411, a1412, a1413, c15, a1501, a1504, a1505, a1506, a1507, a1508, a1509, a1511,
+    a1512, a1513, c16, a1601, a1604, a1605, a1606, a1607, a1608, a1609,
+    a1611, a1612, a1613 = integ.tab.extra
+
+    k11 = f(uprev +
```

Comment:
we can do this for now, but you might want to move these to the interpolation so that they are only computed when required. But I'm not sure if that will cut them out within GPU warps, so maybe it doesn't matter. It's worth experimenting.